src/adapters/serverbid.js

```diff
@@ -44,44 +48,67 @@ const ServerBidAdapter = function ServerBidAdapter() {
 
   baseAdapter.callBids = function(params) {
     if (params && params.bids && utils.isArray(params.bids) && params.bids.length) {
-      const data = {
-        placements: [],
-        time: Date.now(),
-        user: {},
-        url: utils.getTopWindowUrl(),
-        referrer: document.referrer,
-        enableBotFiltering: true,
-        includePricingData: true
-      };
-
-      const bids = params.bids || [];
-      for (let i = 0; i < bids.length; i++) {
-        const bid = bids[i];
-
-        bidIds.push(bid.bidId);
-
-        const bid_data = {
-          networkId: bid.params.networkId,
-          siteId: bid.params.siteId,
-          zoneIds: bid.params.zoneIds,
-          campaignId: bid.params.campaignId,
-          flightId: bid.params.flightId,
-          adId: bid.params.adId,
-          divName: bid.bidId,
-          adTypes: bid.adTypes || getSize(bid.sizes)
+      if (!window.SMARTSYNC) {
+        _callBids(params);
+      } else {
+        window[SMARTSYNC_CALLBACK] = function() {
+          window[SMARTSYNC_CALLBACK] = function() {};
+          _callBids(params);
         };
 
-        if (bid_data.networkId && bid_data.siteId) {
-          data.placements.push(bid_data);
-        }
-      }
+        const siteId = params.bids[0].params.siteId;
+        _appendScript(SMARTSYNC_BASE_URI + '/' + siteId + '.js');
 
-      if (data.placements.length) {
-        ajax(BASE_URI, _responseCallback, JSON.stringify(data), { method: 'POST', withCredentials: true, contentType: 'application/json' });
+        const sstimeout = window.SMARTSYNC_TIMEOUT || ((params.timeout || 500) / 2);
+        setTimeout(function() {
+          var cb = window[SMARTSYNC_CALLBACK];
+          window[SMARTSYNC_CALLBACK] = function() {};
```

Comment:
The reason for this `setTimeout()` call is to ensure that bids are fetched in the case that the callback (`window[SMARTSYNC_CALLBACK]`) is not called by the smartsync script in a timely manner. We also want to make sure that bids aren't fetched more than once. So consider the following cases:

1. smartsync calls the callback promptly: this means that the timeout on line 63 has not yet fired, and we want to fetch bids and then set the callback to a no-op, so we don't fetch bids again when the timeout on line 63 fires.
2. smartsync never calls the callback: this means that the timeout on line 63 will fire, fetch bids, and set the callback to a no-op.
3. smartsync calls the callback after the timeout has fired: in this case we have already fetched bids and we don't want to fetch them again, so the callback must have already been set to a no-op. The code that would have done that  is on line 65.